rbi/core/array.rbi

```diff
@@ -2799,7 +2799,7 @@ class Array < Object
   sig do
     type_parameters(:U).params(
         arg: T::Enumerable[T.type_parameter(:U)],
-        blk: T.proc.params(x: Elem, y: T.nilable(T.type_parameter(:U))).void
+        blk: T.proc.params(x: [Elem, T.nilable(T.type_parameter(:U))]).void
```

Comment:
I don't think that this solves the actual problem either—if you do

```ruby
xs = T.let([1,2,3], T::Array[Integer])
res = xs.zip(xs, xs) do |zipped|
  T.reveal_type(zipped)
end
```

Now `zipped` shows as

```
[Integer, T.nilable(Integer)] (2-tuple)
```

despite the correct type being

```
[Integer, T.nilable(Integer), T.nilable(Integer)] (3-tuple)
```

I would recommend that we split up the change to change the result type to `nil` from the change attempting to fix the argument type yielded to the block.

Off the top of my head, I'm not aware of any current intrinsics which attempt to model this distinction.

The easiest way to work around this in a way that's at least correct if unsatisfying would be to change the RBI like this


```suggestion
        blk: T.proc.params(x: T::Array[T.any(Elem, T.nilable(T.type_parameter(:U)))]).void
```

Possibly with an overload for the case where there is exactly one positional argument to annotate that the block will be yielded exactly two values.